codelab-master/apps/playground/src/app/quiz/containers/introduction/introduction.component.ts

```diff
@@ -0,0 +1,45 @@
+import { ChangeDetectionStrategy, Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+import { QUIZ_DATA } from '@codelab-quiz/shared/quiz-data';
+import { Quiz } from '@codelab-quiz/shared/models/Quiz.model.ts';
+import { QuizService } from '@codelab-quiz/shared/services/quiz.service';
+
+
+@Component({
+  selector: 'codelab-quiz-intro',
+  templateUrl: './introduction.component.html',
+  styleUrls: ['./introduction.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class IntroductionComponent implements OnInit, OnDestroy {
+  quizData: Quiz[] = JSON.parse(JSON.stringify(QUIZ_DATA));
```

Comment:
why do we need to clone it every time?
this is not a great DX< because someone can use it and forget to clone.

If you're mutating it at any point, this indicates that your app design can be improved

Reply:
Ok I changed it to just QUIZ_DATA, I think the reason I used this was for the resetting of the quiz.

Reply:
I believe I changed it in all the initializations, I'll check that the resetting works once the navigation works.